android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -718,13 +730,20 @@ public void remove(TiViewProxy child)
 	@Kroll.method
 	public void removeAllChildren()
 	{
-		if (children != null) {
-			//children might be altered while we loop through it (threading)
-			//so we first copy children as it was when asked to remove all children
-			ArrayList<TiViewProxy> childViews = new ArrayList<TiViewProxy>();
-			childViews.addAll(children);
-			for (TiViewProxy child : childViews) {
-				remove(child);
+		if (children != null && view != null) {
+			View nv = view.getNativeView();
+			if (!(nv instanceof TiCompositeLayout) && nv.getClass().getSimpleName()!="ListViewWrapper") {
+				nv = ((ViewGroup) nv).getChildAt(0);
+			}
+				
+			if (nv instanceof ViewGroup) {
+				ViewGroup vg = (ViewGroup) nv;
+				if (TiApplication.isUIThread()) {
+					vg.removeAllViews();
```

Comment:
I didn't do any memory checks but according to http://stackoverflow.com/a/29942391/5193915 it will null all child views but it won't call release() on all children, which is the other part of handleRemove(). I'll try to see if I can do some memory testing.